spec/services/manage_braintree_donation_spec.rb

```diff
@@ -0,0 +1,112 @@
+require 'rails_helper'
+
+describe ManageBraintreeDonation do
+  # Hey, let's define a new class that makes testing this possible
+
+  class DeepStruct < OpenStruct
```

Comment:
Parsing Braintree results expects a series nested classes which expose particular attributes via a message notation (for instance, `result.amount`). This becomes complicated to mock with a standard instance double, and creating Braintree results requires delving into the Braintree library itself at a level that's honestly untenable. Ruby contains a standard library class called `OpenStruct` which reads in a hash and turns it into an instance which allows for the aforementioned notation. The only problem is, it doesn't support nested hashes. This class is designed (and does) to do that. It's only useful within the context of this spec, so it's defined in the file for this spec.